x11rb-protocol/src/connection/special_events.rs

```diff
@@ -0,0 +1,158 @@
+//! Contains a management strategy for special events.
+
+use std::{convert::TryInto, fmt};
+
+/// A series of selectors, used to classify events as special.
+#[derive(Debug, Default)]
+pub(crate) struct SpecialEventClassifier {
+    /// The event classes.
+    classes: Vec<SpecialEventClass>,
+}
+
+/// A class of event.
+struct SpecialEventClass {
+    /// The number that is used to select the queue.
+    queue_selector: u32,
+    /// The condition for an event being selected.
+    condition: Condition<'static>,
+}
```

Comment:
Do we need this? What about just having something like `special_event_queues: HashMap<u32, (Condition, EventQueue)>` in `Connection`?

Reply:
Two reasons:

1). That would be a pretty complicated type, and "annotating" it like this makes its purpose immediately clear.
2). It allows us to split the special event lookup logic into another file, since the `connection/mod.rs` file is getting pretty long.
3). I’m not using a `HashMap` since the "hot" operation is a linear search of the map, which `HashMap` provides no advantage over a `Vec`. Since the other operations (insert and removal) are cold (probably only called once per program), I decided to use a `Vec` to avoid the map overhead.